[StyleCop]  Add warning fixes generating Breaking Changes

Aliandi opened the pull request:
*Disclaimer*: To fix some SryleCop warnings we might incur in breaking changes. In this case related to protected properties that might be used by end users.
- Rename protected field to match recommended casing

ceciliaavila commented on .NET/Microsoft.Recognizers.Text/Matcher/AcAutomaton.cs:
```diff
@@ -1,98 +1,98 @@
-
-        public AcAutomaton()
-        {
+﻿using System.Collections.Generic;
```
Please, check **AcAutomaton** and **TrieTree**. This appears to be a line ending change.

ceciliaavila replied:
Done!

ceciliaavila reviewed (changes requested):
Please check for line ending problems.

VictorGrycuk reviewed (approved):
LGTM

ceciliaavila reviewed (approved):
LGTM

JuanAr commented on .NET/Microsoft.Recognizers.Text.NumberWithUnit/Parsers/BaseCurrencyParser.cs:
```diff
@@ -65,9 +65,9 @@ private static void ResolveText(List<ParseResult> prs, string source, int bias)
 
         private string GetResolutionStr(object value)
         {
-            return Config.CultureInfo != null ?
```
This is not a breaking change. not that it is incorrect, but it should not be part of this PR.

ceciliaavila replied:
Done. I'll fix this file in another PR.
Thanks!

JuanAr reviewed (changes requested):
Extract the non-breaking change.

ceciliaavila committed:
Fix all StyleCop's warnings in TextMatchers files

- Reorder properties, fields and methods
- Remove trailing whitespace
- Add trailing comma
- Reformat setters and getters

ceciliaavila committed:
Fix all the StyleCop warnings in Parsers

Files:
    -BaseMergedUnitParser.cs
    -INumberWithUnitParserConfiguration.cs
Warnings:
    -SA1400	Class should declare an access modifier
    -SA1401	Field should be private
    -SA1122	Use string.Empty for empty strings

ceciliaavila committed:
Fix all the StyleCop warnings in NumberWithUnitParser

Warnings:
    -SA1401	 Field should be private
    -SA1122	 Use string.Empty for empty strings
    -SA1124	Do not use regions
    -Add file with the class UnitValue
